conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

```diff
@@ -89,29 +89,10 @@ private static void serviceException(HttpServerExchange exchange, ServiceExcepti
 
     private static void qosException(HttpServerExchange exchange, QosException qosException) {
         qosException.accept(QOS_EXCEPTION_HEADERS).accept(exchange);
-        if (qosExceptionHasAdditionalMetadata(qosException)) {
-            log.info("quality-of-service intervention", qosException);
```

Comment:
What if we add a rate limiter here, and log at info ~1 permit per second? That should strike a balance between causing our storage service to spam logs and the current state where these aren’t terribly debuggable.
If that works well enough, I could see removing the additional-metadata check in favor of a pure-ratelimiter approach.

What do you think?

Reply:
That'll make the log much less useful - I'd expect normally we'd have a specific trace(Id) that throws a 429, and we'd miss this stacktrace unless it happens to be one that is able to acquire the rate limiter.

Why is there a concern that this will spam? This will log at most once per request, so will be no worse than request logs.

Reply:
Stack traces are much denser than anything in the request log

Reply:
I don't expect the volume of `QosException` to be that high through.

I agree with what @kevinychen said above - it's very valuable to take a trace ID for a 4xx or 5xx request and be able to easily determine why the request failed.